### PR TITLE
Adding notes describing how to work with Canary

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

File: dev-notes/JavaSDK_With_CanarySubscriptions.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+
+There are two canary regions:
+
+```
+Central US EUAP
+East US 2 EUAP
+```
+
+If you have an Azure subscription whitelisted for Canary then you can use it as usual with canary regions.
```
Comment (praries880) [inline dev-notes/JavaSDK_With_CanarySubscriptions.md]: Suggestion :
It would be nice to make it clear that they need a whitelisted subscription and help the customer with a link to instructions on how to get whitelisted.I would reword this sentence something like :
```suggestion
You need to have a subscription whitelisted to be able to work with Canary regions. Go [here](<link to the site here>) for instructions on how to get a subscription whitelisted.
```

File: dev-notes/JavaSDK_With_CanarySubscriptions.md [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+As usual create credentials instance and use it to initialize Azure client
```
Comment (praries880) [inline dev-notes/JavaSDK_With_CanarySubscriptions.md]:
```suggestion
Create credentials for initializing an azure client instance. You can use the following sample for guidance :
```

File: dev-notes/JavaSDK_With_CanarySubscriptions.md [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+```java
+final String clientId = “<clientId>”;
+final String tenantId = “<tenanId>”;
+final String clientSecret = “<clientSecret>”;
+final String subscriptionId = “<subscriptionId>”; // Regular azure subscription whitelisted for Canary
```
Comment (praries880) [inline dev-notes/JavaSDK_With_CanarySubscriptions.md]:
```suggestion
final String subscriptionId = “<subscriptionId>”; // The azure subscription whitelisted for Canary access
```

File: dev-notes/JavaSDK_With_CanarySubscriptions.md [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+
+```java
+String rgName = "<new-resource-group-name>";
+        
+azure.resourceGroups().define(rgName)
+            .withRegion("Central US EUAP") // Canary region as string
+            .create();
+
+```
+you can also pass strongly typed region:
```
Comment (praries880) [inline dev-notes/JavaSDK_With_CanarySubscriptions.md]:
```suggestion
You can also pass a strongly typed region:
```

File: dev-notes/JavaSDK_With_DogfoodSubscriptions.md [hunk 5]
```diff
@@ -0,0 +1,40 @@
+
+### Using Java SDK with Dogfood subscriptions
+
+If you have a dogfood subscription for ARM endpoint https://api-dogfood.resources.windows-int.net/ then first you will need to build an AzureEnvironment instance for Canary:
```
Comment (praries880) [inline dev-notes/JavaSDK_With_DogfoodSubscriptions.md]: Can you add a link to instructions on how to get a subscription whitelisted for access to the dogfood endpoint?

File: dev-notes/JavaSDK_With_DogfoodSubscriptions.md [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+        this.put("resourceManagerEndpointUrl", "https://api-dogfood.resources.windows-int.net/");
+        this.put("activeDirectoryEndpointUrl", "https://login.windows-ppe.net/");
+        this.put("activeDirectoryResourceId", "https://management.core.windows.net/");
+        this.put("activeDirectoryGraphResourceId", "https://graph.ppe.windows.net/");
+    }
+});
+
+```
+
+Create a credential instance with above environment object and use to initialize Azure client.
```
Comment (praries880) [inline dev-notes/JavaSDK_With_DogfoodSubscriptions.md]:
```suggestion
Create a credential instance with above environment object and use it to initialize an instance of Azure client.
```

File: dev-notes/JavaSDK_With_CanarySubscriptions.md (hunk 1 above)
Comment (anuchandy) [reply dev-notes/JavaSDK_With_CanarySubscriptions.md]: thanks for rewording, I incorporated it but as discussed offline link is not added.

File: dev-notes/JavaSDK_With_CanarySubscriptions.md (hunk 2 above)
Comment (anuchandy) [reply dev-notes/JavaSDK_With_CanarySubscriptions.md]: Done

File: dev-notes/JavaSDK_With_DogfoodSubscriptions.md (hunk 5 above)
Comment (anuchandy) [reply dev-notes/JavaSDK_With_DogfoodSubscriptions.md]: as discussed this is not added intentionally.

File: dev-notes/JavaSDK_With_DogfoodSubscriptions.md (hunk 6 above)
Comment (anuchandy) [reply dev-notes/JavaSDK_With_DogfoodSubscriptions.md]: done

File: dev-notes/JavaSDK_With_CanarySubscriptions.md (hunk 4 above)
Comment (anuchandy) [reply dev-notes/JavaSDK_With_CanarySubscriptions.md]: done
